coins/monero/src/wallet/seed/classic/mod.rs

```diff
@@ -0,0 +1,250 @@
+use std::collections::HashMap;
+
+use crc::{Crc, CRC_32_ISO_HDLC};
+use rand_core::OsRng;
+
+use crate::random_scalar;
+
+use super::{SeedError, LanguageName};
+
+mod languages;
+
+pub const CLASSIC_SEED_LENGTH: usize = 24;
+pub const CLASSIC_SEED_LENGTH_WITH_CHECKSUM: usize = 25;
+const LANGUAGE_WORD_COUNT: usize = 1626;
+
+#[derive(Clone)]
+pub struct Language {
+  word_list: Vec<String>,
+  word_map: HashMap<String, usize>,
+  trimmed_word_map: HashMap<String, usize>,
+  language_name: LanguageName,
+  unique_prefix_length: usize,
+}
+
+impl Language {
+  /// populates the word_map and trimmed_word_maps
+  fn populate_maps(
```

Comment:
Should this be part of a Language constructor so it's impossible to have a Language which doesn't have populated maps?